Library/src/main/java/dev/testify/ScreenshotRule.kt

```diff
@@ -444,29 +448,13 @@ open class ScreenshotRule<T : Activity> @JvmOverloads constructor(
         }
     }
 
-    /**
-     * Test lifecycle method.
-     * Invoked immediately before assertSame and before the activity is launched.
-     */
-    open fun beforeAssertSame() {}
-
-    /**
-     * Test lifecycle method.
-     * Invoked prior to any view modifications and prior to layout inflation.
-     */
-    open fun beforeInitializeView(activity: Activity) {}
-
-    /**
-     * Test lifecycle method.
-     * Invoked after layout inflation and all view modifications have been applied.
-     */
-    open fun afterInitializeView(activity: Activity) {}
+    fun addScreenshotObserver(observer: ScreenshotLifecycle) {
```

Comment:
Non blocking thought, make this could be part of the configuration? Allows for scoping observers to specific tests?

Reply:
🤔 
Very interesting idea. I'll think on the implications of this kind of change, but it might be cleaner & more flexible if it's part of the configuration.

Reply:
I'm also not 100% familiar with how junit executes test cases. Do we get a new rule for each test case or does it get shared across the suite?

Reply:
It's shared within the class. You get a new instance for each test class instance.